drivers/backup/portworx/portworx.go

```diff
@@ -1040,6 +1061,31 @@ func (p *portworx) UpdateOwnershipRule(ctx context.Context, req *api.RuleOwnersh
 	return p.ruleManager.UpdateOwnership(ctx, req)
 }
 
+func (p *portworx) GetBackupUID(ctx context.Context, orgID, backupName string) (string, error) {
+	var backupUID string
+	var totalBackups int
+	bkpEnumerateReq := &api.BackupEnumerateRequest{OrgId: orgID}
+	bkpEnumerateReq.EnumerateOptions = &api.EnumerateOptions{MaxObjects: uint64(enumerateBatchSize), ObjectIndex: 0}
+	for {
+		enumerateRsp, err := p.EnumerateBackup(ctx, bkpEnumerateReq)
+		if err != nil {
+			return backupUID, fmt.Errorf("Failed to enumerate backups for org %s ctx: [%v]", orgID, err)
+		}
+		for _, backup := range enumerateRsp.GetBackups() {
+			if backup.GetName() == backupName {
+				return backup.GetUid(), nil
+			}
+			totalBackups++
+		}
+		if uint64(totalBackups) >= enumerateRsp.GetTotalCount() {
+			break
```

Comment:
there could be a case where the condition on line:1075 never gets satisfied in this case, we might return backupUID as blank. Is this desired behaviour?

Reply:
Yes, it is like a backup is not there and we are trying to find it.

Now if the UID is an empty string, backup inspect will fail with 
"backup object %s UID needs to be provided for inspect".

Previous to the UID fix, when only name was getting passed then also the inspect would have failed but with a different error. 

IMO the function returning empty string is fine, now its upto the caller if it wants to use this empty string or not for further calls.